### PR TITLE
mimirtool: retain old default of blocks_storage.backend

### LINKED ISSUE #1622: mimirtool: Failed to pick the correct backend when running `convert`
#### Describe the bug
I ran the convert tool on my existing app and it picked an incorrect `backend` value post conversions

#### To Reproduce

My config:

```
auth_enabled: false

server:
  http_listen_port: 9009

  # Configure the server to allow messages up to 100MB.
  grpc_server_max_recv_msg_size: 104857600
  grpc_server_max_send_msg_size: 104857600
  grpc_server_max_concurrent_streams: 1000

distributor:
  shard_by_all_labels: true
  pool:
    health_check_ingesters: true

ingester_client:
  grpc_client_config:
    # Configure the client to allow messages up to 100MB.
    max_recv_msg_size: 104857600
    max_send_msg_size: 104857600
    grpc_compression: gzip

ingester:
  lifecycler:
    # The address to advertise for this ingester.  Will be autodiscovered by
    # looking up address on eth0 or en0; can be specified if this fails.
    # address: 127.0.0.1

    # We want to start immediately and flush on shutdown.
    join_after: 0
    min_ready_duration: 0s
    final_sleep: 0s
    num_tokens: 512

    # Use an in memory ring store, so we don't need to launch a Consul.
    ring:
      kvstore:
        store: inmemory
      replication_factor: 1

storage:
  engine: blocks

blocks_storage:
  tsdb:
    dir: /tmp/cortex/tsdb

  bucket_store:
    sync_dir: /tmp/cortex/tsdb-sync

  # You can choose between local storage and Amazon S3, Google GCS and Azure storage. Each option requires additional configuration
  # as shown below. All options can be configured via flags as well which might be handy for secret inputs.
  backend: s3 # s3, gcs, azure or filesystem are valid options
  s3:
    bucket_name: bucket_name
    endpoint: s3.us-east-1.wasabisys.com
    # Configure your S3 credentials below.
    secret_access_key: secretkey
    access_key_id:     accessKeyId
#  gcs:
#    bucket_name: cortex
#    service_account: # if empty or omitted Cortex will use your default service account as per Google's fallback logic
#  azure:
#    account_name:
#    account_key:
#    container_name:
#    endpoint_suffix:
#    max_retries: # Number of retries for recoverable errors (defaults to 20)
# filesystem:
#   dir: ./data/tsdb

compactor:
  data_dir: /tmp/cortex/compactor
  sharding_ring:
    kvstore:
      store: inmemory

frontend_worker:
  match_max_concurrent: true

ruler:
  enable_api: true
  enable_sharding: false

ruler_storage:
  backend: local
  local:
    directory: /tmp/cortex/rules
```

Running this command `./mimirtool config convert --yaml-file=current.yaml --update-defaults --yaml-out=mimir.yaml`

Produced

```
❯ cat mimir.yaml
blocks_storage:
    backend: filesystem
    bucket_store:
        sync_dir: /tmp/cortex/tsdb-sync
    s3:
        access_key_id: accessKeyId
        bucket_name: bucket_name
        endpoint: s3.us-east-1.wasabisys.com
        secret_access_key: secretkey
    tsdb:
        dir: /tmp/cortex/tsdb
compactor:
    data_dir: /tmp/cortex/compactor
    sharding_ring:
        kvstore:
            store: inmemory
distributor:
    pool:
        health_check_ingesters: true
ingester:
    ring:
        final_sleep: 0s
        join_after: 0s
        kvstore:
            store: inmemory
        min_ready_duration: 0s
        num_tokens: 512
        replication_factor: 1
ingester_client:
    grpc_client_config:
        grpc_compression: gzip
        max_recv_msg_size: 104857600
        max_send_msg_size: 104857600
multitenancy_enabled: false
ruler:
    enable_api: true
ruler_storage:
    backend: local
    local:
        directory: /tmp/cortex/rules
server:
    grpc_server_max_concurrent_streams: 1000
    grpc_server_max_recv_msg_size: 104857600
    grpc_server_max_send_msg_size: 104857600
    http_listen_port: 9009
```

Which shows the `blocks_storage: backend: filesystem` which should be `s3`

#### Environment
```
❯ ./mimirtool version
Mimirtool, version 2.0.0 (branch: release-2.0, revision: 9fd2da5d3)
  go version:       go1.17.8
  platform:         linux/amd64
checking latest version... A newer version of mimirtool is available, please update to mimir-2.0.0
```

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
#### What this PR does

When passing `--update-defaults` to mimirtool config convert and the
value of blocks_storage.backend is using the old default of "s3", the
value in the resulting config is updated to "filesystem." It is better
if the value is not updated.

#### Which issue(s) this PR fixes or relates to

Fixes #1622 

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [na] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pstibrany — COMMENTED** [review]
What about other storages? (Alertmanager, ruler) I would expect the update path to be the same for all of them (if alertmanager / ruler were used in the first place).

**Comment by pstibrany** [comment]
I think this now deserves a CHANGELOG entry too.

**Comment by dimitarvdimitrov** [comment]
nice call @pstibrany. I've added `(ruler|alertmanager)_storage.backend` to the list of retained old defaults in https://github.com/grafana/mimir/pull/1626/commits/740878d0722073cc56fb5d8e1e57924483a27399 and added a changelog entry in https://github.com/grafana/mimir/pull/1626/commits/fbbc35514b14ca5fdc87c5d81dd4cacd7145144c 

I also added some docs to mention the special behaviour for these defaults in https://github.com/grafana/mimir/pull/1626/commits/a075db07946d5d1a2864fc36bfa5a237da3a6531. Also added `type/docs` label to get some advice from Docs squad on that paragraph.

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 1]
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
...
+
+This command is designed to aid a migration from Cortex to Grafana Mimir. There are changes to the default values of
+some configuration parameters in Mimir v2.0.0 that we think you would not want to make as part of this migration. These
+parameters are:
+
+- `blocks_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `ruler_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `alertmanager_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `server.http_listen_port` - Cortex default is `80`, Mimir default is `8080`
+- (GEM only) `auth.type` - Cortex default is `trust`, Mimir default is `enterprise`
```
Comment (pstibrany) [inline docs/sources/operators-guide/tools/mimirtool.md:668]: Nit: It's GEM default, not Mimir.

**Review by pstibrany — APPROVED** [review]
Looks good to me. There is no good / easy way checking if alertmanager and ruler were used before (in Cortex) :(

**Review by pracucci — APPROVED** [review]
I'm not much familiar with the config conversion tool, but to my understanding the changes make sense.

**Comment by dimitarvdimitrov** [comment]
thank you for reviewing. For visibility - I'm waiting for the docs squad to get to this PR so they can review the added docs to mimirtool. 

There is other work related to this that I'd like to do - some filesystem paths (like `/data/compactor` changed default values, and we can employ a similar approach as in this PR to provide a smoother migration.

I want these to be included in the 2.1 release, so if I don't get a docs review in the next week. We can extract the docs in a seperate PR and move on with the code changes

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 2]
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
 | `-v`, `--verbose`    | If you set this flag, the CLI flags and YAML paths from the old configuration that do not exist in the new configuration are printed to `stderr`. This flag also prints default values that have changed between the old and the new configuration. |
 | `--gem`              | If you set this flag, the tool will convert from Grafana Metrics Enterprise (GEM) v1.7.x to v2.0.0.                                                                                                                                                 |
 
+##### A note on default values
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:658]:
```suggestion
##### Changes to default values
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 3]
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
...
+
+This command is designed to aid a migration from Cortex to Grafana Mimir. There are changes to the default values of
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:660]:
```suggestion
The `mimirtool` command helps you migrate from Cortex to Grafana Mimir. There are changes to the default values of
```
I assume we are talking about the `mimirtool` here.

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 4]
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
...
+some configuration parameters in Mimir v2.0.0 that we think you would not want to make as part of this migration. These
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:661]:
```suggestion
some configuration parameters in Mimir v2.0.0 that you might not want to use as part of this migration:
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 5]
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
...
+parameters are:
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:662]:
```suggestion
```

File: docs/sources/operators-guide/tools/mimirtool.md [hunk 6]
```diff
@@ -655,6 +655,22 @@ It supports converting both CLI flags and [YAML configuration files]({{< relref
...
+
+- `blocks_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `ruler_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `alertmanager_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
+- `server.http_listen_port` - Cortex default is `80`, Mimir default is `8080`
+- (GEM only) `auth.type` - GEM 1.x default is `trust`, 2.x default is `enterprise`
+
+This command will output the Cortex default value even when the configuration parameter is not explicitly set in the input
+configuration. If the Cortex default value is explicitly set in the input configuration, and you have provided
+the `--update-defaults` flag, the command will not update the value to the Mimir default.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/tools/mimirtool.md:672]:
```suggestion
The `mimirtool` command outputs the Cortex default value even when the configuration parameter is not explicitly set in the input
configuration. If in the input configuration you explicitly set the Cortex default value, and you have provide
the `--update-defaults` flag, the `mimirtool` command will not update the value to the Mimir default.
```
Please vet for technical accuracy.

**Review by osg-grafana — COMMENTED** [review]
Unblocking with feedback, which I would like in part to be technically vetted for accuracy.

**Review by osg-grafana — APPROVED** [review]
See previous comment. :)

File: docs/sources/operators-guide/tools/mimirtool.md (hunk 6 above)
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/tools/mimirtool.md]: changed in cecbdd27675f13e9375b09289347cb4186623398, can you take another look?

**Pull request merged**